File: kq-projection-dotnet/kq-projection/Common/Enums/EventType.cs
namespace KQ_Projection.Common.Enums;

using System.Runtime.Serialization;
using System.Text.Json.Serialization;

[JsonConverter(typeof(JsonStringEnumConverter))]
public enum EventType
{
    [EnumMember(Value = "GAME_STARTED_EVENT")]
    GAME_STARTED_EVENT = 0,

    [EnumMember(Value = "PIECE_ADDED_EVENT")]
    PIECE_ADDED_EVENT = 1,

    [EnumMember(Value = "PIECE_REMOVED_EVENT")]
    PIECE_REMOVED_EVENT = 2,

    [EnumMember(Value = "REVERT_MOVES_EVENT")]
    REVERT_MOVES_EVENT = 3,

    [EnumMember(Value = "SAVE_CURRENT_STATE_EVENT")]
    SAVE_CURRENT_STATE_EVENT = 4,

    [EnumMember(Value = "STARTED_FROM_SAVED_STATE_EVENT")]
    STARTED_FROM_SAVED_STATE_EVENT = 5,

    [EnumMember(Value = "GAME_PAUSED_EVENT")]
    GAME_PAUSED_EVENT = 6,

    [EnumMember(Value = "GAME_RESUMED_EVENT")]
    GAME_RESUMED_EVENT = 7,

    [EnumMember(Value = "EXCEPTIONAL_EVENT")]
    EXCEPTIONAL_EVENT = 8,

    [EnumMember(Value = "DISCARD_GAME_EVENT")]
    DISCARD_GAME_EVENT = 9
}

File: java-apps/kq-common/src/main/java/com/kafkaquest/kq/common/util/models/events/enums/EventType.java
package com.kafkaquest.kq.common.util.models.events.enums;

public enum EventType {

    GAME_STARTED_EVENT(0),
    PIECE_ADDED_EVENT(1),
    PIECE_REMOVED_EVENT(2),
    REVERT_MOVES_EVENT(3),
    SAVE_CURRENT_STATE_EVENT(4),
    STARTED_FROM_SAVED_STATE_EVENT(5),
    GAME_PAUSED_EVENT(6),
    GAME_RESUMED_EVENT(7),
    EXCEPTIONAL_EVENT(8),
    DISCARD_GAME_EVENT(9);

    private final int eventNumber;

    EventType(int eventNumber) {
        this.eventNumber = eventNumber;
    }

    public int getEventNumber() {
        return eventNumber;
    }

}
